# Practice Assignments - Insurance Data

## Table of Contents


This notebook will be used for 4 practice assignments:
- [📝 Loading Data (Practice)](#loading)
- [📝 Slicing (Practice)](#slicing)
- [📝 Filtering (Practice)](#filtering)


Click on the assignment names above to jump to the corresponding headers.  



___




Note that:
  - The first assignment (Loading Data) must be completed before the other three assignments may be completed.

  - We recommend you do each of the remaining sections as you encounter them on the Learning Platform.


<a name="loading"> </a>
# 📝 Loading Data (Practice)


[Insurance Data](https://drive.google.com/file/d/1-7mcBguuzAV7JWVu2XCpgAO-fO6X2cl0/view?usp=sharing)

1. Open the link and save the shared file to your Drive in the week 2 data folder.

2. Mount your drive

3. Find and copy the file path of the insurance_mod.csv file

In [ ]:
from google.colab import drive
drive.mount('/content/drive')

Mounted at /content/drive


In [ ]:
# Save the copied file path as a variable
# Surround the path with quotes
fpath  ="/content/insurance_mod.csv"

#fpath = "/content/mydrive/LiveClassMaterial/Foundational-course/week/date../insurance_mod.csv"

4. Load in the file as a Pandas dataframe named df_insurance


In [ ]:
# Include needed import(s)
import pandas as pd


# Load in the file as a Pandas dataframe named df_insurance
df_insurance = pd.read_csv("/content/insurance_mod.csv")

In [ ]:
df_insurance

age     sex     bmi  children  smoker     region  charges
0      19  female  27.900         0       1  southwest  16885.0
1      18    male  33.770         1       0  southeast   1726.0
2      28    male  33.000         3       0  southeast   4449.0
3      33    male  22.705         0       0  northwest  21984.0
4      32    male  28.880         0       0  northwest   3867.0
...   ...     ...     ...       ...     ...        ...      ...
1333   50    male  30.970         3       0  northwest  10601.0
1334   18  female  31.920         0       0  northeast   2206.0
1335   18  female  36.850         0       0  southeast   1630.0
1336   21  female  25.800         0       0  southwest   2008.0
1337   61  female  29.070         0       1  northwest  29141.0

[1338 rows x 7 columns]

5. Preview the first 5 rows with .head()

In [ ]:
# Preview the first 5 rows with .head()
df_insurance.head(10)

age     sex     bmi  children  smoker     region  charges
0   19  female  27.900         0       1  southwest  16885.0
1   18    male  33.770         1       0  southeast   1726.0
2   28    male  33.000         3       0  southeast   4449.0
3   33    male  22.705         0       0  northwest  21984.0
4   32    male  28.880         0       0  northwest   3867.0
5   31  female  25.740         0       0  southeast   3757.0
6   46  female  33.440         1       0  southeast   8241.0
7   37  female  27.740         3       0  northwest   7282.0
8   37    male  29.830         2       0  northeast   6406.0
9   60  female  25.840         0       0  northwest  28923.0

6. Preview the last 5 rows with .tail()

In [ ]:
# Preview the last 5 rows with .tail()
df_insurance.tail(10)

age     sex     bmi  children  smoker     region  charges
1328   23  female  24.225         2       0  northeast  22396.0
1329   52    male  38.600         2       0  southwest  10325.0
1330   57  female  25.740         2       0  southeast  12629.0
1331   23  female  33.400         0       0  southwest  10796.0
1332   52  female  44.700         3       0  southwest  11412.0
1333   50    male  30.970         3       0  northwest  10601.0
1334   18  female  31.920         0       0  northeast   2206.0
1335   18  female  36.850         0       0  southeast   1630.0
1336   21  female  25.800         0       0  southwest   2008.0
1337   61  female  29.070         0       1  northwest  29141.0

7.  Obtain summary information with .info()


In [ ]:
# Obtain summary information with .info()
df_insurance.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 1338 entries, 0 to 1337
Data columns (total 7 columns):
 #   Column    Non-Null Count  Dtype  
---  ------    --------------  -----  
 0   age       1338 non-null   int64  
 1   sex       1338 non-null   object 
 2   bmi       1338 non-null   float64
 3   children  1338 non-null   int64  
 4   smoker    1338 non-null   int64  
 5   region    1338 non-null   object 
 6   charges   1338 non-null   float64
dtypes: float64(2), int64(3), object(2)
memory usage: 73.3+ KB


8. Use the output of .info() to answer the following in a text cell below:


a) How many rows are there?
- There are 1,338 rows.


b) How many columns are there?
- There are 7 columns

<a name="slicing"> </a>
# 📝 Slicing (Practice)


1. Slice the row for the customer at index 50 as a Series


In [ ]:
# Slice just the row for the customer at index 50
# Use single brackets so the output is a Pandas Series
df_insurance.loc[50]


age                18
sex            female
bmi            35.625
children            0
smoker              0
region      northeast
charges        2211.0
Name: 50, dtype: object

2. Slice the row for the customer at index 50 as a DataFrame


In [ ]:
# Slice just the row for the customer at index 50
# Use double brackets so the output is a Pandas Dataframe
df_insurance.loc[[50]]

age     sex     bmi  children  smoker     region  charges
50   18  female  35.625         0       0  northeast   2211.0

3. Find out if the customer at index 20 was a male or female (by slicing index 20 and column "Sex")


In [ ]:
# find out if the customer at index 20 was a male or female by slicing index 20 and column "Sex"
df_insurance.loc[20, 'sex']

'female'

4. Slice the entire smoker column using square brackets


In [ ]:
# Slice the entire smoker column using square brackets
df_insurance['smoker']

0       1
1       0
2       0
3       0
4       0
       ..
1333    0
1334    0
1335    0
1336    0
1337    1
Name: smoker, Length: 1338, dtype: int64

5. Slice the entire smoker column using .loc


In [ ]:
# Slice the entire smoker column using .loc
df_insurance.loc[:,'smoker']

0       1
1       0
2       0
3       0
4       0
       ..
1333    0
1334    0
1335    0
1336    0
1337    1
Name: smoker, Length: 1338, dtype: int64

6. Slice the entire smoker column using either method, but this time save it as a variable called "smoker"


In [ ]:
# Slice the entire smoker column using either method, but this time save it as a variable called "smoker"
smoker = df_insurance['smoker']
smoker

0       1
1       0
2       0
3       0
4       0
       ..
1333    0
1334    0
1335    0
1336    0
1337    1
Name: smoker, Length: 1338, dtype: int64

- Calculate the sum of the smoker variable to see how many of the customers smoke


In [ ]:
# To see how many of the customers smoke, get the sum
smoker.sum()

274

- Calculate the mean of the smoker variable using .mean() to see the ratio of smokers in the dataset


In [ ]:
# To get the ratio use .mean()
smoker.mean()

0.20478325859491778

In [ ]:
smoker.median()

0.0

<a name="filtering"> </a>
# 📝 Filtering (Practice)

1. Create a filter variable called under_30_filter for rows where the age is less than 30


In [ ]:
# Create a filter called under_30_filter for the rows where the age is less than 30
under_30_filter = df_insurance['age'] < 30

In [ ]:
# View the under_30_filter to see the boolean index
under_30_filter

0        True
1        True
2        True
3       False
4       False
        ...  
1333    False
1334     True
1335     True
1336     True
1337    False
Name: age, Length: 1338, dtype: bool

In [ ]:
# Get the sum of the true values for the under_30_filter
# This is the number of customers who are under the age of 30
under_30_filter.sum()

417

In [ ]:
# # Use the filter to view only rows with customers who are under 30
df_insurance[under_30_filter]

age     sex     bmi  children  smoker     region  charges
0      19  female  27.900         0       1  southwest  16885.0
1      18    male  33.770         1       0  southeast   1726.0
2      28    male  33.000         3       0  southeast   4449.0
10     25    male  26.220         0       0  northeast   2721.0
12     23    male  34.400         0       0  southwest   1827.0
...   ...     ...     ...       ...     ...        ...      ...
1328   23  female  24.225         2       0  northeast  22396.0
1331   23  female  33.400         0       0  southwest  10796.0
1334   18  female  31.920         0       0  northeast   2206.0
1335   18  female  36.850         0       0  southeast   1630.0
1336   21  female  25.800         0       0  southwest   2008.0

[417 rows x 7 columns]

In [ ]:
age_filter = df_insurance['age']==30
age_filter.sum()

27

2. Create a filter for males


In [ ]:
# Create a filter for males called male_filter
male_filter = df_insurance['sex']=='male'

In [ ]:
# View the filter to see the boolean index
male_filter

0       False
1        True
2        True
3        True
4        True
        ...  
1333     True
1334    False
1335    False
1336    False
1337    False
Name: sex, Length: 1338, dtype: bool

In [ ]:
# Obtain the sum of True values in the male_filter
# This is the number of customers who are male
male_filter.sum()

676

In [ ]:
# Use the filter to view only rows with customers who are male
df_insurance[male_filter]

age   sex     bmi  children  smoker     region  charges
1      18  male  33.770         1       0  southeast   1726.0
2      28  male  33.000         3       0  southeast   4449.0
3      33  male  22.705         0       0  northwest  21984.0
4      32  male  28.880         0       0  northwest   3867.0
8      37  male  29.830         2       0  northeast   6406.0
...   ...   ...     ...       ...     ...        ...      ...
1324   31  male  25.935         1       0  northwest   4240.0
1325   61  male  33.535         0       0  northeast  13143.0
1327   51  male  30.030         1       0  southeast   9378.0
1329   52  male  38.600         2       0  southwest  10325.0
1333   50  male  30.970         3       0  northwest  10601.0

[676 rows x 7 columns]

3. Combine the under_30_filter and the male_filter to create a single filter called under_30_male_filter


In [ ]:
# Combine the under_30_filter and the male_filter to create a single filter called under_30_male_filter
under_30_male_filter = male_filter & under_30_filter

In [ ]:
# Display the boolean index of the under_30_male_filter
under_30_male_filter

0       False
1        True
2        True
3       False
4       False
        ...  
1333    False
1334    False
1335    False
1336    False
1337    False
Length: 1338, dtype: bool

In [ ]:
# Get the sum of the under_30_male_filter True values
# This is the number of customers who are both under 30 and male
under_30_male_filter.sum()

216

In [ ]:
# Use the filter to view only rows with customers who are both under 30 and male
df_insurance[under_30_male_filter]

age   sex     bmi  children  smoker     region  charges
1      18  male  33.770         1       0  southeast   1726.0
2      28  male  33.000         3       0  southeast   4449.0
10     25  male  26.220         0       0  northeast   2721.0
12     23  male  34.400         0       0  southwest   1827.0
14     27  male  42.130         0       1  southeast  39612.0
...   ...   ...     ...       ...     ...        ...      ...
1292   21  male  23.210         0       0  southeast   1515.0
1295   20  male  22.000         1       0  southwest   1965.0
1296   18  male  26.125         0       0  northeast   1709.0
1315   18  male  28.310         1       0  northeast  11272.0
1317   18  male  53.130         0       0  southeast   1163.0

[216 rows x 7 columns]

4. Create a filter for regions that contain south using .str.contains


In [ ]:
# Create a filter for regions that contain south using .str.contains
filter_south = df_insurance['region'].str.contains('south')

In [ ]:
# Use the filter to view only rows with regions containing "south"
df_insurance[filter_south]

age     sex    bmi  children  smoker     region  charges
0      19  female  27.90         0       1  southwest  16885.0
1      18    male  33.77         1       0  southeast   1726.0
2      28    male  33.00         3       0  southeast   4449.0
5      31  female  25.74         0       0  southeast   3757.0
6      46  female  33.44         1       0  southeast   8241.0
...   ...     ...    ...       ...     ...        ...      ...
1330   57  female  25.74         2       0  southeast  12629.0
1331   23  female  33.40         0       0  southwest  10796.0
1332   52  female  44.70         3       0  southwest  11412.0
1335   18  female  36.85         0       0  southeast   1630.0
1336   21  female  25.80         0       0  southwest   2008.0

[689 rows x 7 columns]

In [ ]:
southeast_filter = df_insurance['region']=='southwest'

southeast_filter.sum()

325